test/feature/dynamic_types/DynamicTypesTests.cpp

```diff
@@ -0,0 +1,7670 @@
+    ASSERT_TRUE(seq_data);
+    ASSERT_EQ(seq_data->get_int32_values(good_seq, 0), RETCODE_OK);
+    ASSERT_EQ(good_seq, Int32Seq({1, 0}));
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+
+    seq_data = data->loan_value(0);
+    ASSERT_TRUE(seq_data);
+    ASSERT_NE(seq_data->set_uint32_values(0, {1, 2}), RETCODE_OK);
+    ASSERT_EQ(seq_data->set_int32_values(0, {1, 2}), RETCODE_OK);
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(1);
+    ASSERT_TRUE(seq_data);
+    ASSERT_NE(seq_data->set_uint32_value(1, 1), RETCODE_OK);
+    ASSERT_EQ(seq_data->set_int32_value(1, 1), RETCODE_OK);
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(2);
+    ASSERT_TRUE(seq_data);
+    ASSERT_NE(seq_data->set_uint32_values(0, {3, 4}), RETCODE_OK);
+    ASSERT_EQ(seq_data->set_int32_values(0, {3, 4}), RETCODE_OK);
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(3);
+    ASSERT_TRUE(seq_data);
+    ASSERT_NE(seq_data->set_uint32_value(0, 1), RETCODE_OK);
+    ASSERT_EQ(seq_data->set_int32_value(0, 1), RETCODE_OK);
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+
+    ASSERT_EQ(RETCODE_OK, data->clear_value(3));
+    ASSERT_EQ(RETCODE_BAD_PARAMETER, data->clear_value(100));
+    ASSERT_EQ(4u, data->get_item_count());
+    seq_data = data->loan_value(0);
+    ASSERT_TRUE(seq_data);
+    ASSERT_EQ(seq_data->get_int32_values(good_seq, 0), RETCODE_OK);
+    ASSERT_EQ(good_seq, Int32Seq({1, 2}));
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(1);
+    ASSERT_TRUE(seq_data);
+    ASSERT_EQ(seq_data->get_int32_values(good_seq, 0), RETCODE_OK);
+    ASSERT_EQ(good_seq, Int32Seq({0, 1}));
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(2);
+    ASSERT_TRUE(seq_data);
+    ASSERT_EQ(seq_data->get_int32_values(good_seq, 0), RETCODE_OK);
+    ASSERT_EQ(good_seq, Int32Seq({3, 4}));
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+    seq_data = data->loan_value(3);
+    ASSERT_TRUE(seq_data);
+    ASSERT_EQ(seq_data->get_int32_values(good_seq, 0), RETCODE_OK);
+    ASSERT_EQ(good_seq, Int32Seq({0, 0}));
+    ASSERT_EQ(RETCODE_OK, data->return_loaned_value(seq_data));
+
+    DynamicDataFactory::get_instance()->delete_data(data);
+}
+
+TEST_F(DynamicTypesTests, DynamicType_map)
+{
+    DynamicTypeBuilderFactory::_ref_type factory {DynamicTypeBuilderFactory::get_instance()};
+
+    const uint32_t map_length {2};
+
+    DynamicTypeBuilder::_ref_type builder {factory->create_map_type(
+                                               factory->get_primitive_type(eprosima::fastdds::dds::TK_INT32),
+                                               factory->get_primitive_type(eprosima::fastdds::dds::TK_INT32),
+                                               map_length)};
```

Comment:
Please add a case where the map is created using the TypeDescriptor. Also, I am missing the negative tests for this API:

* Zero length
* Non-supported map key (including wstrings)

Also the negative tests checking inconsistent map TypeDescriptors:

* Non nil base type and/or discriminator type
* Zero bound
* Nil element type and/or key element type